tesseract_ros/tesseract_rviz/src/render_tools/environment_widget.cpp

```diff
@@ -365,10 +365,10 @@ bool EnvironmentWidget::applyEnvironmentCommands(const std::vector<tesseract_msg
         std::vector<std::string> child_link_names =
             tesseract_->getEnvironment()->getSceneGraph()->getLinkChildrenNames(remove_joint->child_link_name);
 
-        if (!visualization_->removeLink(command.remove_link))
+        if (!visualization_->removeJoint(joints[0]->getName()))
           return false;
 
-        if (!visualization_->removeJoint(joints[0]->getName()))
+        if (!visualization_->removeLink(remove_joint->child_link_name))
           return false;
```

Comment:
If this is the case then line 334 and 331 should also be switched. Do you mind making the change in this PR since they are related?